compiler/hash-ast/src/resolve.rs

```diff
@@ -129,20 +129,20 @@ where
                     |elapsed| println!("ast: {:.2?}", elapsed),
                 );
 
-                if let Ok(module) = import_node {
-                    // @@Cleanup:
-                    // @@Hack: we still need to add the contents of the file into the file map whilst the parser
-                    //         is parsing. If we don't do this, the reporting crate won't be able to access the
-                    //         contents of the file for reporting purposes.
-                    ctx.module_builder.add_contents(
-                        import_index,
-                        resolved_import_path,
-                        import_source,
-                    );
-                    ctx.module_builder.add_module_at(import_index, module); // Add the import to modules
+                // @@Cleanup:
+                // @@Hack: we still need to add the contents of the file into the file map whilst the parser
+                //         is parsing. If we don't do this, the reporting crate won't be able to access the
+                //         contents of the file for reporting purposes.
+                ctx.module_builder
+                    .add_contents(import_index, resolved_import_path, import_source);
+
+                match import_node {
+                    Ok(module) => {
+                        ctx.module_builder.add_module_at(import_index, module); // Add the import to modules
+                        Ok(())
+                    }
+                    Err(err) => Err(err),
```

Comment:
? syntax could be used here.

Reply:
Well unfortunately the call still has to be made to add the contents of the file into the module builder:
```rs
 ctx.module_builder.add_contents(import_index, resolved_import_path, import_source);
```

If the call is made before, the string is moved and the borrow checking complains about this.

Reply:
```rs
import_node.map(|module| ctx.module_builder.add_module_at(import_index, module))
```
Doesn't this do the same thing?

Reply:
`add_module_at` doesn't add the contents of the source for the very reason.